src/Libraries/RevitNodes/Elements/InternalUtilities/ElementUtils.cs

```diff
@@ -187,38 +194,142 @@ public static void SetParameterValue(Autodesk.Revit.DB.Parameter param, bool val
         public static double GetConvertedParameterValue(Autodesk.Revit.DB.Parameter param, double value)
         {
             var paramType = param.Definition.ParameterType;
-
-            if (Element.IsConvertableParameterType(paramType))
+            UnitType unitType;
+            if (ParamTypeMap.TryGetValue(paramType, out unitType))
             {
-                return value * UnitConverter.DynamoToHostFactor(ParameterTypeToUnitType(paramType));
+                return value * UnitConverter.DynamoToHostFactor(unitType);
             }
-
-            return value;
-        }
-
-        [SupressImportIntoVM]
-        internal static UnitType ParameterTypeToUnitType(ParameterType parameterType)
-        {
-            switch (parameterType)
+            else
             {
-                case ParameterType.Length:
-                    return UnitType.UT_Length;
-                case ParameterType.Area:
-                    return UnitType.UT_Area;
-                case ParameterType.Volume:
-                    return UnitType.UT_Volume;
-                case ParameterType.Angle:
-                    return UnitType.UT_Angle;
-                case ParameterType.Slope:
-                    return UnitType.UT_Slope;
-                case ParameterType.Currency:
-                    return UnitType.UT_Currency;
-                case ParameterType.MassDensity:
-                    return UnitType.UT_MassDensity;
-                default:
-                    throw new Exception(Properties.Resources.UnitTypeConversionError);
+                return value;
             }
         }
+
+        private static Dictionary<ParameterType, UnitType> ParamTypeMap = new Dictionary<ParameterType, UnitType>()
+        {
+            { ParameterType.Number, UnitType.UT_Number},
+            { ParameterType.Length, UnitType.UT_Length},
+            { ParameterType.Area, UnitType.UT_Area},
+            { ParameterType.Volume, UnitType.UT_Volume},
+            { ParameterType.Angle, UnitType.UT_Angle},
+            { ParameterType.Force, UnitType.UT_Force},
```

Comment:
@mjkkirschner I believe that a Dictionary will be a little faster than a switch statement for this case. It also eliminates a need for the initial test, since we can just use `TryGetValue`. What you think?